stdlib/runtime/exception.gr

```diff
@@ -29,10 +29,18 @@ export let dangerouslyRegisterBasePrinter = (f) => {
     }
     current = next
   }
+  // Leaks memory (cyclic dep. with Memory) :(
+  //Memory.decRef(WasmI32.fromGrain(dangerouslyRegisterBasePrinter))
+  //Memory.decRef(WasmI32.fromGrain(f))
+  void
 }
 
-export let dangerouslyRegisterPrinter = (f) => {
+export let rec dangerouslyRegisterPrinter = (f) => {
   printers = WasmI32.fromGrain((f, printers))
+  // Leaks memory (cyclic dep. with Memory) :(
+  //Memory.decRef(WasmI32.fromGrain(dangerouslyRegisterPrinter))
+  //Memory.decRef(WasmI32.fromGrain(f))
```

Comment:
```suggestion
  // We don't decRef the closure or arguments here to avoid a cyclic dep. on Memory.
  // This is fine, as this function is only called seldomly.
```